src/blockdata/opcodes.rs

```diff
@@ -674,77 +674,62 @@ impl All {
-                // 76 opcodes
-                } else if self.code <= all::OP_PUSHBYTES_75.code {
-                    Class::PushBytes(self.code as u32)
-                // 61 opcodes
-                } else {
-                    Class::Ordinary(Ordinary::try_from_all(self).unwrap())
-                }
+        use self::all::*;
+        match (self, ctx) {
+            // 3 opcodes illegal in all contexts
+            (OP_VERIF, _) | (OP_VERNOTIF, _) | (OP_INVALIDOPCODE, _) => Class::IllegalOp,
+
+            // 15 opcodes illegal in Legacy context
+            (OP_CAT, ctx) | (OP_SUBSTR, ctx) 
+            | (OP_LEFT, ctx) | (OP_RIGHT, ctx) 
+            | (OP_INVERT, ctx) 
+            | (OP_AND, ctx) | (OP_OR, ctx) | (OP_XOR, ctx)
+            | (OP_2MUL, ctx) | (OP_2DIV, ctx)
+            | (OP_MUL, ctx) | (OP_DIV, ctx) | (OP_MOD, ctx)
+            | (OP_LSHIFT, ctx) | (OP_RSHIFT, ctx) if ctx == ClassifyContext::Legacy => Class::IllegalOp,
+
+            // 87 opcodes of SuccessOp class only in TapScript context
+            (op, ClassifyContext::TapScript)
+            if op.code == 80 || op.code == 98 ||
+                (op.code >= 126 && op.code <= 129) ||
+                (op.code >= 131 && op.code <= 134) ||
+                (op.code >= 137 && op.code <= 138) ||
+                (op.code >= 141 && op.code <= 142) ||
+                (op.code >= 149 && op.code <= 153) ||
+                (op.code >= 187 && op.code <= 254) => Class::SuccessOp,
+
+            // 11 opcodes of NoOp class
+            (OP_NOP, _) => Class::NoOp,
+            (op, _) if op.code >= OP_NOP1.code && op.code <= OP_NOP10.code => Class::NoOp,
+
+            // 1 opcode for `OP_RETURN`
+            (OP_RETURN, _) => Class::ReturnOp,
+
+            // 4 opcodes operating equally to `OP_RETURN` only in Legacy context
+            (OP_RESERVED, ctx)
+            | (OP_RESERVED1, ctx) | (OP_RESERVED2, ctx)
+            | (OP_VER, ctx) if ctx == ClassifyContext::Legacy => Class::ReturnOp,
+
+            // 71 opcodes operating equally to `OP_RETURN` only in Legacy context
+            (op, ClassifyContext::Legacy) if op.code >= OP_CHECKSIGADD.code => Class::ReturnOp,
+
+            // 2 opcodes operating equally to `OP_RETURN` only in TapScript context
+            (OP_CHECKMULTISIG, ClassifyContext::TapScript)
+            | (OP_CHECKMULTISIGVERIFY, ClassifyContext::TapScript) => Class::ReturnOp,
+
+            // 1 opcode of PushNum class
+            (OP_PUSHNUM_NEG1, _) => Class::PushNum(-1),
+
+            // 16 opcodes of PushNum class
+            (op, _) if op.code >= OP_PUSHNUM_1.code && op.code <= OP_PUSHNUM_16.code => {
+                Class::PushNum(1 + self.code as i32 - OP_PUSHNUM_1.code as i32)
+            },
+
+            // 76 opcodes of PushBytes class
+            (op, _) if op.code <= OP_PUSHBYTES_75.code => Class::PushBytes(self.code as u32),
+
+            // 61 opcodes of Ordinary class
```

Comment:
minor nit(don't fix unless needed): This was also an error in my code. 61 opcodes for Legacy and 60 for Tapscript.

Reply:
Fixed since it was needed to to end-line space character removal requirement :D

Reply:
Thanks :D